### ISSUE TITLE
Observables arrays don't appear to observe correctly

### ISSUE THREAD

**Issue by Dairnon** [nx-js/observer-util#35]
I have an observables array that I observe to listen to changes.

On size change I update the information based on if it changed size.

Increasing the array with push seems to work fine. It reacts properly.

However splicing or poping it (reducing it's size) calls the reaction twice... This second reaction returns the same length array but with the value set to null in the position of where the pop happened (and I assume where any element removed)

This is causing me a lot of issues in being able to observe arrays changes....Ive had to resort to manually doing reactions...tho I'd prefer if it was automatic based on observe.

I read somewhere in another issue that arrays do weird things with proxy and observables...so I was curious if this could be fixed.

**Comment by solkimicreb** [comment]
This is correct behavior, but you probably want to use a scheduler. See the [related docs](https://github.com/nx-js/observer-util#reaction-scheduling), make sure to check all 3 expandable examples.

The observer-util schedules a reaction on every single related data change. In case of complex array mutations a lot of changes happen in the background. You should rarely use this lib without passing a custom scheduler to it. Schedulers can batch and delay reactions. The last example in the docs uses the [queue-util](https://github.com/nx-js/queue-util) as a scheduler, I suggest you to play with it (especially with the ciritical prio queue). The queue util lets you combine auto priority queues and manual control over when to execute reactions.

I hope this helps, let me know if you need more info.

**Comment by Dairnon** [comment]
Alrighty, I'll play around with combining the debugger with the queue's and see if I can replicate mobx array observing!!!

I'll let u know if I can do it with my tests... and if not let you know...
